### PR TITLE
refactor caddy-tailscale logic

### PR DESCRIPTION
**Pull request by willnorris**
This is admittedly a pretty sizeable refactor, but is almost exclusively renames, code movement, and documentation.  There *should* be no logic changes.  There is technically one breaking change in renaming `TSApp.Servers` to `TSApp.Nodes`, but only for folks using the JSON config directly (caddyfile doesn't use the name).

The refactoring in this change includes:
 - Move TailscaleAuth logic into auth.go
 - Move all TSApp logic into app.go (including caddyfile parsing)
 - Rename "server" to "node" throughout. This aligns better with Tailscale terminology, and is reflective of the fact that nodes can also just be used as proxy transports, in which case they are not acting as servers at all.
 - Generally prefer referring to a node's "name" than "host". While this name is still used as the default hostname for the node, I would expect that to change with a future iteration of #18.
 - add godocs throughout

### CODE REVIEW

**Review by mholt — COMMENTED** [review]
Neat! I didn't read it too thoroughly yet but with a quick scan I don't see anything that caught my attention other than a typo :) (I don't know much/anything about TS-domain stuff so I was just looking at the Caddy stuff)

File: transport.go [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package tscaddy
 
+// transpoart.go contains the TailscaleCaddyTransport module.
```
[the remark this answers is not in the document]
Comment (willnorris) [reply transport.go]: done

**Comment by willnorris** [comment]
awesome, thanks for giving it a once-over!

File: transport.go [hunk 2]
```diff
@@ -21,7 +33,8 @@ func (t *TailscaleCaddyTransport) UnmarshalCaddyfile(d *caddyfile.Dispenser) err
 func (t *TailscaleCaddyTransport) Provision(ctx caddy.Context) error {
 	t.logger = ctx.Logger()
 
-	s, err := getServer(ctx, "caddy-tsnet-client:80")
+	// TODO(will): allow users to specify a node name used to lookup that node's config in TSApp.
+	s, err := getNode(ctx, "caddy-tsnet-client:80")
```
Comment (clly) [inline transport.go:37]: Should this be just `caddy-tsnet-client`?

**Review by clly — APPROVED** [review]
Looks like a really clean refactor!

File: transport.go (hunk 2 above)
Comment (willnorris) [reply transport.go:37]: oh, good catch! I remember seeing that, and forgot to fix it

**Pull request merged**